pkg/config/repoconfig/repoconfig.go

```diff
@@ -15,14 +15,43 @@ const (
 )
 
 type BotConfig struct {
-	LabelApproved *struct {
-		Approvals *int
-		Label     *string
-	} `yaml:"label-approved"`
+	LabelApproved *LabelApprovedConfig `yaml:"label-approved"`
 }
 
-func Read(gitRepo *git.Git, sha string) (*BotConfig, error) {
+type LabelApprovedConfig struct {
+	Approvals *int
+	Label     *string
+}
+
+func (c *BotConfig) HasLabelApproved() bool {
+	return c.LabelApproved != nil
+}
+
+func (c *BotConfig) GetLabelApproved() *LabelApprovedConfig {
```

Comment:
```suggestion
func (c *BotConfig) LabelApproved() *LabelApprovedConfig {
```

This is more "go"

Reply:
Oh, I didn't know we could shadow the attribute name with a label name, are we sure this is possible?. Let me try.

Reply:
```
Step 8/16 : RUN go build -o main pkg/main/main.go
 ---> Running in 7b96866b3934
# github.com/submariner-io/pr-brancher-webhook/pkg/config/repoconfig
pkg/config/repoconfig/repoconfig.go:30:6: type BotConfig has both field and method named LabelApproved
```
We can't but, let me wrap the BotConfig as BotConfigYAML